DOCS-2902: Make machine api edits from qa (#3505)
viamrobotics/docs: docs/appendix/apis/robot.md

---
title: "Manage Machines with Viam's Machine Management API"
linkTitle: "Machine Management"
weight: 20
type: "docs"
description: "How to use the Machine API to monitor and manage your machines."
tags: ["robot state", "sdk", "apis", "robot api"]
aliases:
  - /program/apis/robot/
  - /build/program/apis/robot/
---

> 

Author: ```suggestion
```

---
title: "Manage Machines with Viam's Machine Management API"
linkTitle: "Machine Management"
weight: 20
type: "docs"
description: "How to use the Machine API to monitor and manage your machines."
tags: ["robot state", "sdk", "apis", "robot api"]
aliases:
  - /program/apis/robot/
  - /build/program/apis/robot/
---

The _machine API_ is the application programming interface that manages each of your machines running `viam-server`.

> Use the machien API to connect to your machine from within a supported [Viam SDK](/appendix/apis/), and send commands remotely.

Author: ```suggestion
Use the machine API to connect to your machine from within a supported [Viam SDK](/appendix/apis/), and send commands remotely.
```

from viam.robot.client import RobotClient

async def connect() -> RobotClient:
    opts = RobotClient.Options(
        disable_sessions=True,
        dial_options=DialOptions(timeout=10)).with_api_key(
            # Replace "<API-KEY>" (including brackets) with your API key
            api_key='<API-KEY>',
            # Replace "<API-KEY-ID>" (including brackets) with your API key
            # ID
            api_key_id='<API-KEY-ID>'
        )
    return await RobotClient.at_address(

>       address='ADDRESS FROM CODE SAMPLE TAB OF VIAM APP',

Author: ```suggestion
      address='ADDRESS FROM CONNECT TAB OF VIAM APP',
```
